Documentation/workflow/commit-messages.md

```diff
@@ -0,0 +1,335 @@
+```
+
+then this can be converted into the bulleted list:
+
+```
+  * org/repo@deadbeef: [ExampleComponent] Example summary (#1234)
+```
+
+The `git log old-commit...dependency-commit` output should also be reviewed,
+and if any commits mention that any bugs are fixed, the fixed bugs should be
+converted into [`Context: `](#Context) links.
+
+If more than a dozen commits are included in the bump, it is too verbose to
+mention all the commits.  Mentioned bug fixes should still be mentioned as
+`Context: ` links, and the bulleted list can be skipped.
+
+For bumps that do not involve GitHub submodule bumps, the summary should be:
+
+```
+Bump to [Dependency Name] [Dependency Version]
+```
+
+If possible, provide a `Changes: ` link to the changelog of the dependency
+which includes the relevant changes.
+
+
+[compare-commits]: https://help.github.com/en/github/committing-changes-to-your-project/comparing-commits
+
+
+
+## Bug Fix
+
+Bug fixes shouldn't mention a bug number in the commit summary, but instead
+should summarize the nature of the fix:
+
+```
+[Component] Short summary of the change
+```
+
+The commit body should contain a [`Fixes: `](#Fixes) section mentioning the bug
+URL, if possible.
+
+The bug should be described: what was required to encounter the bug?  Were any
+MSBuild properties required to be set to particular values?
+
+*Were any error messages printed*?  *All* relevant error messages should be
+present within the commit message body, as this makes it easier to verify if
+a new bug report has already been fixed.
+
+*Do not rely on* the `Fixes: ` URL to provide the above information.  The URL
+may become inaccessible in the future, or the URL contents may be difficult to
+follow, e.g. there was lots of "back and forth" between the submitter and those
+responsible for fixing the bug.  *Summarize* the issue.
+
+A description of the fix should be provided, as well as any other relevant
+information.
+
+
+<a name="Sections" />
+
+# Sections
+
+The commit message may have the following sections or labels, in this order:
+
+  * [`Changes`](#Changes)
+  * [`Fixes`](#Fixes)
+  * [`Context`](#Context)
+
+
+<a name="Changes" />
+
+## Changes
+
+When bumping a dependency such as a git submodule or a NuGet package reference,
+the `Changes: ` line should be present and is a link to the fixes included
+within the bump.
+
+If the bump contains a "small" number of changes -- less than a dozen? -- then
+the commit message can *also* contain a bulleted list of all the commits within
+the bump.
+
+
+<a name="Fixes" />
+
+## Fixes
+
+`Fixes: ` is used to indicate that a bug located at a specific URL is fixed by
+the commit.  *Full* URLs are to be used, *not* GitHub abbreviations, as just
+because GitHub is used this week doesn't mean it will continue to use GitHub
+for the infinite future.  (GitHub is already Xamarin.Android's *third* bug
+repo, after `bugzilla.novell.com` and `bugzilla.xamarin.com`...)
+
+Links to private bugs may be used.
```

Comment:
Is it worth mentioning kzu's Chrome extension?
https://github.com/kzu/azdo
https://chrome.google.com/webstore/detail/azdo-linkinator/copkjnnnmemkbfmolfacgmfiecfjopkk

I use this to generate nice, short URLS, for both Azure DevOps links & Developer Community.

Like this is a short link to a build: https://build.azdo.io/3599640